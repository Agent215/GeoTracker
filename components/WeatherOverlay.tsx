import { UrlTile } from "react-native-maps";
import React from "react"


/**
 * This is not an elegent solution 
 * The urlTemplate prop in the UrlTile prop seems to not rerender with a variable or state passed
 * This is a hacky work around. 
 */
const WeatherOverlay = (props) => {

    let weather;
    let clouds = (
        <UrlTile
            key={props.date + props.category}
            urlTemplate={'https://tile.openweathermap.org/map/clouds_new/{z}/{x}/{y}.png?appid=0a5664abda83ce0f6668159611a1daa6'}
            maximumZ={19}
            flipY={false}

        />

    );
    let temp = (
        <UrlTile
            key={props.date + props.category}
            urlTemplate={'https://tile.openweathermap.org/map/temp_new/{z}/{x}/{y}.png?appid=0a5664abda83ce0f6668159611a1daa6'}
            maximumZ={19}
            flipY={false}

        />

    );
    let precip = (
        <UrlTile
            key={props.date + props.category}
            urlTemplate={'https://tile.openweathermap.org/map/precipitation_new/{z}/{x}/{y}.png?appid=0a5664abda83ce0f6668159611a1daa6'}
            maximumZ={19}
            flipY={false}

        />

    );
    let pressure = (
        <UrlTile
            key={props.date + props.category}
            urlTemplate={'https://tile.openweathermap.org/map/pressure_new/{z}/{x}/{y}.png?appid=0a5664abda83ce0f6668159611a1daa6'}
            maximumZ={19}
            flipY={false}

        />

    );
    let wind = (
        <UrlTile
            key={props.date + props.category}
            urlTemplate={'https://tile.openweathermap.org/map/wind_new/{z}/{x}/{y}.png?appid=0a5664abda83ce0f6668159611a1daa6'}
            maximumZ={19}
            flipY={false}

        />

    );
    let none = (
        null


    );

    if (props.category == undefined) weather = none;
    if (props.category == "") weather = none;
    if (props.category == "clouds") weather = clouds;
    if (props.category == "temp") weather = temp;
    if (props.category == "precipitation") weather = precip;
    if (props.category == "wind") weather = wind;
    if (props.category == "pressure") weather = pressure;
    if (props.category == "none") weather = none;

    if (props.gibsVisible) weather = none;

    return (
        weather)

}


export default WeatherOverlay;