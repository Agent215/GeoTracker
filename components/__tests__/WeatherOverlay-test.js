import * as React from 'react';
import renderer from 'react-test-renderer';
import WeatherOverlay from '../../components/WeatherOverlay';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

//array of categories to test randomly
const categories = ["clouds", "temp", "precipitation", "wind", "pressure"]

describe('<WeatherOverlay />', () => {

    it(`WeatherOverlay renders Correctly on Load`, () => {

        const weatherOverlayComponent = renderer.create(<WeatherOverlay />).toJSON()    
        expect(weatherOverlayComponent).toMatchSnapshot();
        });

    it('WeatherOverlay is null if no weather to be displayed', () => {

        const weatherOverlayComponent = renderer.create(<WeatherOverlay category={"none"} gibsVisible={true} />).toJSON();
    })

    it(`WeatherOverlay is not null if weather to be displayed`, () => {

        const weatherOverlayComponent = renderer.create(<WeatherOverlay category={ categories[Math.floor(Math.random() * categories.length)] } gibsVisible={false} />).toJSON()   
        expect(weatherOverlayComponent).not.toBeNull();
        });

    it('WeatherOVerlay is null if gibs is visible', () => {

        const weatherOverlayComponent = renderer.create(<WeatherOverlay category={"clouds"} gibsVisible={true} />).toJSON()
        expect(weatherOverlayComponent).toBeNull();
    })
})