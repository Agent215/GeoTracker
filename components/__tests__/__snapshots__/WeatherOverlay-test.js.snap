// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherOverlay /> WeatherOverlay renders Correctly on Load 1`] = `null`;
